src/codeu/chat/common/Bot.java

```diff
@@ -0,0 +1,13 @@
+package codeu.chat.common;
+
+public abstract class Bot {
+
+    abstract String reactTo(String command, String[] args);
+    abstract String onAdd();  // what bot does when it is first added
+    public String getName()
```

Comment:
I updated it to have comments to explain the logic behind this. In addition, it should return the name of the class, so it can be hardcoded, so it doesn't need to be final